proto/cache.proto

```diff
@@ -46,6 +47,64 @@ message GetCacheScoreCardRequest {
   // A page token returned from the previous response, or an empty string
   // initially.
   string page_token = 3;
+
+  message Filter {
+    // Field mask selecting filters to be applied.
+    google.protobuf.FieldMask mask = 1;
+
+    // Return only results with this cache type.
+    CacheType cache_type = 2;
+
+    // Return only results with this request type.
+    RequestType request_type = 3;
+
+    // ResponseType represents a category of RPC response codes.
+    enum ResponseType {
+      UNKNOWN_RESPONSE_TYPE = 0;
+      // Show only successful responses (OK response).
+      OK = 1;
+      // Show only NOT_FOUND responses.
+      NOT_FOUND = 2;
+      // Show only error responses (excluding NOT_FOUND).
+      ERROR = 3;
+    }
```

Comment:
Should this be at the top level too?

Could we use a builtin grpc status proto for this...?

Reply:
moved to the top-level -- resolved offline, but for future readers: this proto is used because we want the user to be able to filter NOT_FOUND and "other" errors separately.